composer/algorithms/factorize/README.md

```diff
@@ -0,0 +1,96 @@
+# ➗ Factorize
+
+
+[\[How to Use\]](#how-to-use) - [\[Suggested Hyperparameters\]](#suggested-hyperparameters) - [\[Technical Details\]](#technical-details) - [\[Attribution\]](#attribution)
+
+ `Computer Vision`  `Natural Language Processing`
+
+Factorize splits a large linear or convolutional layer into two smaller ones that compute a similar function via factorization.
+This can be applied to models for both computer vision and natural language processing.
+
+| ![Factorize](factorize-no-caption.png) |
+|:--:
+|*The weights `W` of a 2D convolutional layer with `k x k` filters, `c` input chanels, and `d` output channels (a) is factorized into two smaller convolutions (b) with weights `W'` and `p` with `d'` intermediate channels. The first convolution uses the original filter size but produces only `d'` channels. The second convolution has `1 x 1` filters and produces the original `d` output channels but has only `d'` input channels. This changes the complexity from O(k<sup>2</sup>cd) to O(k<sup>2</sup>cd') + O(d'd). This image is a Figure in [https://ieeexplore.ieee.org/abstract/document/7332968](Zhang et al. (2015)).*|
+
+## How to Use
+
+### Functional Interface
+
+TODO(DAVIS): FIX
+
+```python
+def training_loop(model, train_loader):
+  opt = torch.optim.Adam(model.parameters())
+  loss_fn = F.cross_entropy
+  model.train()
+  
+  for epoch in range(num_epochs):
+      for X, y in train_loader:
+          y_hat = model(X)
+          loss = loss_fn(y_hat, y)
+          loss.backward()
+          opt.step()
+          opt.zero_grad()
```

Comment:
```suggestion
from composer import functional as cf

def training_loop(model, train_loader):
    opt = torch.optim.Adam(model.parameters())

    # only need to pass in opt if apply_blurpool is used after optimizer creation
    # otherwise only the model needs to be passed in
    cf.apply_factorization(model,
                           factorize_convs=True,
                           factorize_linears=True,
                           min_channels=512,
                           latent_channels=0.25,
                           min_features=512,
                           latent_features=0.25,
                           optimizers=opt)

    loss_fn = F.cross_entropy
    model.train()

    for epoch in range(num_epochs):
        for X, y in train_loader:
            y_hat = model(X)
            loss = loss_fn(y_hat, y)
            loss.backward()
            opt.step()
            opt.zero_grad()
```